src/hotspot/cpu/riscv/macroAssembler_riscv.hpp

```diff
@@ -376,8 +376,18 @@ class MacroAssembler: public Assembler {
     return ((predecessor & 0x3) << 2) | (successor & 0x3);
   }
 
+  void fence(uint32_t predecessor, uint32_t successor) {
+    if (UseZtso) {
+      // do not emit fence if it's not at least a StoreLoad fence
```

Comment:
Could you improve the comments with some examples, next code is hard to read  if ( not  (a and b)) then ..